### PR TITLE
`Introduction` and `dependencies` docs

### PR DESCRIPTION
**Pull request by kasperski95**
- updated docs structure to slug names
- updated edit url
- added `Introduction` doc
- added `Dependencies` doc

New:
- updated home page

### CODE REVIEW

File: website/docs/tutorials/guides/dependencies-management.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+| Format                                | Example                                           |
+| ------------------------------------- | ------------------------------------------------- |
+| `GITHUB_ACCOUNT_NAME/REPO_NAME[@TAG]` | `OpenZeppelin/cairo-contracts@0.1.0`              |
+| `URL_TO_THE_REPOSITORY`               | `https://github.com/OpenZeppelin/cairo-contracts` |
+| `SSH_URI`                             | `git@github.com:OpenZeppelin/cairo-contracts.git` |
+
+
+### Aliases
+
+Protostar supports installing dependencies under a different name. This allows you to resolve a name conflict, in case of two GitHub users use the same name for their library. In order to install a package under a custom name, run `protostar install EXTERNAL_DEPENDENCY_REFERENCE --name CUSTOM_NAME`.
```
Comment (MaksymilianDemitraszek) [inline website/docs/tutorials/guides/dependencies-management.md:47]: From a reader perspective it there arises a question, how can I use aliases for upgrading and removing. Some example would be nice too.

File: website/docs/tutorials/guides/dependencies-management.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+$ tree -L 2
+.
+├── lib
+│   └── cairo_integer_types
+├── protostar.toml
+├── src
+│   └── main.cairo
+└── tests
+    └── test_main.cairo
+```
```
Comment (MaksymilianDemitraszek) [inline website/docs/tutorials/guides/dependencies-management.md:32]: I am not sure if this example output is necessary, I think just a command is more than enough. The protostar itself should provide sufficient feedback about the success.

File: website/docs/tutorials/01-introduction.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+After reading this guides, you will know how to do the following:
+
+- install and upgrade Protostar
+- initialize new Protostar project
+- configure `protostar.toml`
+- adapt existing Cairo project to Protostar project
+- compile project
+- [add, remove, and remove dependencies](/docs/tutorials/Guides/dependencies-management)
+- test contracts with the help of cheatcodes
```
Comment (MaksymilianDemitraszek) [inline website/docs/tutorials/01-introduction.md:31]: I would replace it with a table of contents with good section names. Serves the same purpose but would be more useful for a reader. Would make docs easier to use as a reference not a book which you read from the beginning to the end.

File: website/docs/tutorials/guides/dependencies-management.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
+---
+sidebar_label: Dependencies (2 min)
+---
+
+# Dependencies
+
+Protostar manages dependencies (packages) using [git submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules) ([similarly to Foundry](https://onbjerg.github.io/foundry-book/projects/dependencies.html)). This is the reason Protostar expects [git](https://git-scm.com/) to be installed. Keep this in mind, especially if your project uses git submodules.
+
+:::note
+Using Git submodules as a foundation for package management is not an ideal approach. Therefore, [Starkware](https://starkware.co/) plans to create a proper package manager.
```
Comment (MaksymilianDemitraszek) [inline website/docs/tutorials/guides/dependencies-management.md:10]:
```suggestion
Using Git submodules as a foundation for package management is not an ideal approach. Therefore, [Starkware](https://starkware.co/) plans to create a proper package manager, which we are planning to integrate into protostar in the future.
```

File: website/docs/tutorials/guides/dependencies-management.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+:::
+
+
+Protostar updates dependencies in the following ways:
+- updating to recent tag, if dependency's repository uses [tags](https://git-scm.com/book/en/v2/Git-Basics-Tagging)
+- updating to recent commit, if dependency's repository doesn't use tags
+
+## Removing dependencies
+
+To remove a dependency, run `protostar remove LOCAL_DEPENDENCY_REFERENCE`.
```
Comment (MaksymilianDemitraszek) [inline website/docs/tutorials/guides/dependencies-management.md:66]: Would be nice to have an example

File: website/docs/tutorials/guides/dependencies-management.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+- a single dependency, run `protostar update LOCAL_DEPENDENCY_REFERENCE`
+
+:::info
+**LOCAL_DEPENDENCY_REFERENCE** is any reference shown in [External Dependency Reference Formats](#external-dependency-reference-formats) or the name of a directory storing that dependency.
+:::
+
+
+Protostar updates dependencies in the following ways:
+- updating to recent tag, if dependency's repository uses [tags](https://git-scm.com/book/en/v2/Git-Basics-Tagging)
+- updating to recent commit, if dependency's repository doesn't use tags
```
Comment (MaksymilianDemitraszek) [inline website/docs/tutorials/guides/dependencies-management.md:62]: Would rephrase it more into a story than a list. 

> Protostar uses the recent commit from a provided branch.
> <example>
> If you want you can point to a specific tag using ...
> <example>

Or something simmilar and more examples, this explanations seems a bit cryptic.

Also can I use different branches?

File: website/docs/tutorials/01-introduction.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+sidebar_label: Introduction (1 min)
+---
+
+# Introduction
+
+Protostar is a StarkNet smart contract development [toolchain](https://en.wikipedia.org/wiki/Toolchain), which helps you with the following tasks:
+
+- [dependencies management](/docs/tutorials/Guides/dependencies-management)
+- project compilation
+- testing your contract
```
Comment (MaksymilianDemitraszek) [inline website/docs/tutorials/01-introduction.md:11]:
```suggestion
- testing contracts
```

**Review by MaksymilianDemitraszek — CHANGES_REQUESTED** [review]
Amazing job!!! Some questions and thoughts in the comments.

File: website/docs/tutorials/guides/dependencies-management.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
+---
+sidebar_label: Dependencies (2 min)
+---
+
+# Dependencies
+
+Protostar manages dependencies (packages) using [git submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules) ([similarly to Foundry](https://onbjerg.github.io/foundry-book/projects/dependencies.html)). This is the reason Protostar expects [git](https://git-scm.com/) to be installed. Keep this in mind, especially if your project uses git submodules.
```
[the remark this answers is not in the document]
Comment (kasperski95) [reply website/docs/tutorials/guides/dependencies-management.md:7]: I had 'why' initially, but LanguageTool suggested removing it in picky-mode. I think both versions are correct.

File: website/docs/tutorials/guides/dependencies-management.md [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+
+:::note
+Using Git submodules as a foundation for package management is not an ideal approach. Therefore, [Starkware](https://starkware.co/) plans to create a proper package manager.
+:::
+
+## Adding a dependency
+
+To add a dependency, run `protostar install` from the [working directory](https://en.wikipedia.org/wiki/Working_directory) of your project:
```
[the remark this answers is not in the document]
Comment (kasperski95) [reply website/docs/tutorials/guides/dependencies-management.md:15]: Removing it doesn't provide any value. Maybe some readers have no CS background.

https://en.wikipedia.org/wiki/Curse_of_knowledge

File: website/docs/tutorials/01-introduction.md (hunk 3 above)
Comment (kasperski95) [reply website/docs/tutorials/01-introduction.md:31]: There's already a sidebar on the left side and document summary on the right.  I am not sure whether adding such TOC is possible in docusaurous.  TOC lists everything, learning objectives focuses on the most important skills.

Also, if that would be the case, Google would replace "Learning objectives" with TOC
https://developers.google.com/tech-writing/one

File: website/docs/tutorials/guides/dependencies-management.md (hunk 4 above)
Comment (kasperski95) [reply website/docs/tutorials/guides/dependencies-management.md:10]: Do we?

File: website/docs/tutorials/guides/dependencies-management.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (kasperski95) [reply website/docs/tutorials/guides/dependencies-management.md:15]: I'll change the sentence though.

[the remark this answers is not in the document]
Comment (MaksymilianDemitraszek) [reply website/docs/tutorials/guides/dependencies-management.md:15]: Yes they might not have such knowledge, but they probably have an intuitive understanding of a concept if they managed to install a cli. That's the reason I suggested current directory which is more intuitive term.

[the remark this answers is not in the document]
Comment (kasperski95) [reply website/docs/tutorials/guides/dependencies-management.md:15]: Yep, there's no point of introducing this term.

File: website/docs/tutorials/guides/dependencies-management.md (hunk 4 above)
Comment (MaksymilianDemitraszek) [reply website/docs/tutorials/guides/dependencies-management.md:10]: That's a good question 🤔 . Maybe leave it as it was

File: website/docs/tutorials/01-introduction.md [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+
+:::note
+Protostar takes its inspiration from [Foundry](https://onbjerg.github.io/foundry-book/index.html).
+:::
+
+## Target audience
+
+You have basic knowledge about [Cairo and StarkNet](https://www.cairo-lang.org/docs/).
```
[the remark this answers is not in the document]
Comment (kasperski95) [reply website/docs/tutorials/01-introduction.md]: I changed it a bit. I am not sure why you want to add another sentence when it doesn't provide any new information.

File: website/docs/tutorials/guides/dependencies-management.md [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+
+
+## Updating dependencies
+
+To update:
+
+- a single dependency, run `protostar update LOCAL_DEPENDENCY_REFERENCE`
+- all dependencies, run `protostar update`
+
+`LOCAL_DEPENDENCY_REFERENCE` is a directory name of dependency for example:
```
Comment (MaksymilianDemitraszek) [inline website/docs/tutorials/guides/dependencies-management.md:71]:
```suggestion
`LOCAL_DEPENDENCY_REFERENCE` is a directory name of a dependency, for example:
```

File: website/docs/tutorials/guides/dependencies-management.md [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+$ protostar install
+09:37:42 [INFO] Installing cairo_contracts (https://github.com/OpenZeppelin/cairo-contracts)
+```
+
+
+## Updating dependencies
+
+To update:
+
+- a single dependency, run `protostar update LOCAL_DEPENDENCY_REFERENCE`
```
Comment (MaksymilianDemitraszek) [inline website/docs/tutorials/guides/dependencies-management.md:68]:
```suggestion
- a single dependency, run `protostar update LOCAL_DEPENDENCY_REFERENCE/EXTERNAL_DEPENDENCY_REFERENCE`
```

Comment (MaksymilianDemitraszek) [inline website/docs/tutorials/guides/dependencies-management.md:68]:
```suggestion
- a single dependency, run `protostar update LOCAL_DEPENDENCY_REFERENCE/EXTERNAL_DEPENDENCY_REFERENCE`
```

File: website/docs/tutorials/guides/dependencies-management.md [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+`LOCAL_DEPENDENCY_REFERENCE` is a directory name of dependency for example:
+
+```console title="Updating a previously installed dependency."
+$ protostar update cairo_contracts
+10:03:52 [INFO] Package already up to date.
+```
+
+:::note
+You can use [`EXTERNAL_DEPENDENCY_REFERENCE`](#external-dependency-reference-formats) in place of `LOCAL_DEPENDENCY_REFERENCE`.
+:::
```
Comment (MaksymilianDemitraszek) [inline website/docs/tutorials/guides/dependencies-management.md:80]: Example for that would be nice, something like:

You can also use EXTERNAL_DEPENDENCY_REFERENCE to update 

```console title="Updating a previously installed dependency."
$ protostar update https://github.com/OpenZeppelin/cairo-contracts
10:03:52 [INFO] Package already up to date.
```

Comment (MaksymilianDemitraszek) [inline website/docs/tutorials/guides/dependencies-management.md:80]: Example for that would be nice, something like:

You can also use EXTERNAL_DEPENDENCY_REFERENCE to update 

```console title="Updating a previously installed dependency."
$ protostar update https://github.com/OpenZeppelin/cairo-contracts
10:03:52 [INFO] Package already up to date.
```

File: website/docs/tutorials/guides/dependencies-management.md (hunk 12 above)
Comment (MaksymilianDemitraszek) [reply website/docs/tutorials/guides/dependencies-management.md:68]: Otherwise could be slightly misleading, WDYT?

**Review by MaksymilianDemitraszek — APPROVED** [review]
LGTM

File: website/docs/tutorials/guides/dependencies-management.md (hunk 12 above)
Comment (kasperski95) [reply website/docs/tutorials/guides/dependencies-management.md]: Yep, it's better.

File: website/docs/tutorials/guides/dependencies-management.md (hunk 13 above)
Comment (kasperski95) [reply website/docs/tutorials/guides/dependencies-management.md]: I removed this note. There's no need for an example because usage of EXTERNAL_DEPENDENCY_REFERENCE is demonstrated in a section before.
